packages/@aws-cdk/aws-apigateway/test/base-path-mapping.test.ts

```diff
@@ -0,0 +1,83 @@
+import '@aws-cdk/assert/jest';
+import * as acm from '@aws-cdk/aws-certificatemanager';
+import * as cdk from '@aws-cdk/core';
+import * as apigw from '../lib';
+
+describe('BasePathMapping', () => {
+  test('default setup', () => {
+    // GIVEN
+    const stack = new cdk.Stack();
+    const api = new apigw.RestApi(stack, 'MyApi');
+    api.root.addMethod('GET'); // api must have atleast one method.
+    const domain = new apigw.DomainName(stack, 'MyDomain', {
+      domainName: 'example.com',
+      certificate: acm.Certificate.fromCertificateArn(stack, 'cert', 'arn:aws:acm:us-east-1:1111111:certificate/11-3336f1-44483d-adc7-9cd375c5169d'),
+      endpointType: apigw.EndpointType.REGIONAL,
+    });
+
+    // WHEN
+    new apigw.BasePathMapping(stack, 'MyBasePath', {
+      restApi: api,
+      domainName: domain,
+    });
+
+    // THEN
+    expect(stack).toHaveResource('AWS::ApiGateway::BasePathMapping', {
+      DomainName: { Ref: 'MyDomainE4943FBC' },
+      RestApiId: { Ref: 'MyApi49610EDF' },
+    });
+  });
+
+  test('specify basePath property', () => {
```

Comment:
Thanks for adding these tests. 

Can you also add a test case where basepath has an illegal character and assert that the expected error is produced?

Reply:
@nija-at  I've added the requested test case.